### ISSUE TITLE
Retry Failed Tx gets stuck on Customize Gas

### ISSUE THREAD

**Issue by evvvritt** [MetaMask/metamask-extension#8288]
**Describe the bug**
When clicking the "Retry Transaction" button on a failed Tx in the History feed, a "Customize Gas" screen appears but the user remains stuck there and must reopen the extension. Clicking "Save" or "Close" does nothing.

**To Reproduce**

1. Open MetaMask
2. Open a Failed Tx in History list
3. Click "Retry Transaction" button, with the refresh-icon
4. [Customize Gas screen appears]
5. Click "Close" or "Save", does nothing

**Expected behavior**
"Close" should close the dialog
"Save" should resubmit the Transaction(?)

**Browser details (please complete the following information):**
 - OS: macOS Mojave 10.14.6
 - Browser: Chrome Version 80.0.3987.149 (Official Build) (64-bit)
 - MetaMask Version 7.7.8

**Additional Information**
Failed Tx Info:
```
Amount: 0 ETH
Gas Limit (Units): 23300
Gas Used (Units): 23300
Gas Price (GWEI): 5.6
Total: 0.00013 ETH | $0.02 USD
```

**Comment by samajammin** [comment]
I believe I'm having the same issue here. Screen recording:
https://share.getcloudapp.com/o0ugdnGW

Same issue with the "Basic" view - changing from Slow/Average/Fast & clicking save does nothing.

No console errors.

**Comment by Gudahtt** [comment]
Thanks for the report! I suspect this was caused by the retry button being mistakenly shown for an on-chain failure. We don't currently have a way of retrying those yet, so the retry attempt failed.

This should be fixed as of #9506 - the retry button will now be hidden in cases such as this. Unfortunately there is no built-in way to retry in these cases - you'll have to manually retry instead.

This should be included in the next release. Please let us know if you're still experiencing this issue in v8.1.0 or above, once it has been published!
